File: 49. Group Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        lookup = defaultdict(list)
        for s in strs:
            key = "".join(sorted(list(s)))
            lookup[key].append(s)
            
        output = []
        for l in lookup.values():
            output.append(l)    
            
        return output
        

File: 49. Group Anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const storedStrs = strs.map(w=> w.split('').sort().join(''))
    console.log(storedStrs)
    let obj = {}
    for(let i=0;i<strs.length;i++){
        if(!obj[storedStrs[i]]){
            obj[storedStrs[i]]    = [strs[i]]
        }
        else{
            obj[storedStrs[i]].push(strs[i])
        }
    }
    return Object.values(obj)
};